### ISSUE TITLE
msgpuck

### ISSUE THREAD

**Issue by justinmk** [tarruda/libmpack#1]
Just found this, it seems to have similar goals as libmpack: https://github.com/tarantool/msgpuck . It has a `mpstream-ext-add` branch so I guess that means extension types might not be supported in master. 

Thought it might be useful.

**Comment by tarruda** [comment]
Thanks for mentioning

Msgpuck seems to be a more limited version of libmpack's core module, it only deals with msgpack tokens and doesn't offer an API to incrementally construct arbitrary object graphs from msgpack streams.

It also keeps no state between API calls, making it more cumbersome to parse msgpack streams split across multiple buffers, which is how data arrives in the network(no guarantee that the whole stream will arrive in a contiguous chunk).

In other words, with msgpuck you have to allocate temporary memory and copy chunks to form a complete msgpack payload, and with libmpack you simply call the same functions and it will incrementally tokenize/parse streams, keeping state between calls. So even if data arrives in 1-byte chunks, the tokenization/parsing will work the same as if the msgpack payload was in a single contiguous buffer, with no backtracking.

**Comment by justinmk** [comment]
> with libmpack you simply call the same functions and it will incrementally tokenize/parse streams, keeping state between calls.  So even if data arrives in 1-byte chunks, the tokenization/parsing will work the same as if the msgpack payload was in a single contiguous buffer, with no backtracking.

Cool. Another thing to mention in the readme?

**Comment by tarruda** [comment]
> Cool. Another thing to mention in the readme?

I plan to when I write API documentation. I did not mention before because this is already a feature of msgpack-c, and I focused mostly on advantages.
